### PR TITLE
Improve DG additions

### PR DESCRIPTION
**Pull request by ZechariahTan**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -260,6 +250,14 @@ From the diagram illustrated above:
 ### FindPool feature
 This feature allows users to find a pool that contains a passenger with a provided keyword in their name.
 
+Design considerations include the `findPool` command being able to be used in conjunction with the `pool` or `unpool`  command. Due to the limitations of the current iteration of GME, each passenger can only belong to one pool. As the number of passengers and pools increases, it may be hard to keep track of the passengers that exist in pools and those that dont, thus, the user might attempt to create a new pool with a passenger and be informed that that passenger already exists in a pool. The `findPool` feature thus allows the user to find the pool that the passenger already exists in to either confirm if they are in the correctly assigned pool, or `unpool` them if they are not. The `findPool` command works similarly to the `find` command, except that it currently only supports the use of the name prefix: "/n"
```
Comment (lyueyang) [inline docs/DeveloperGuide.md]: dont is missing an apostrophe

Comment (lyueyang) [inline docs/DeveloperGuide.md]: and also _be informed that that passenger already exists in a pool_

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -260,6 +250,14 @@ From the diagram illustrated above:
...
+
+The activity diagram below encapsulates the user workflow of attempting to pool a passenger, finding out that the passenger already exists in a pool, and using the `findPool` command to rectify the issue:
+
+![Activity Diagram for a user using Unpool](images/UnpoolActivityDiagram.png)
```
Comment (chewterence) [inline docs/DeveloperGuide.md]: unpool acitivity diagram is in findpool feature. perhaps rename the diagram so not to confuse future developers?
"Activity Diagram for a user using findpool with unpool"
maybe emphasize that the focus is on findpool here

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (lyueyang) [inline docs/DeveloperGuide.md]: the name prefix should also be `n/` instead of `/n`

**Review by lyueyang — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (chewterence) [reply docs/DeveloperGuide.md]: use "do not" or "unable to" don't is for conversations

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -250,11 +250,11 @@ From the diagram illustrated above:
...
+Design considerations include the `findPool` command being able to be used in conjunction with the `pool` or `unpool`  command. Due to the limitations of the current iteration of GME, each passenger can only belong to one pool. As the number of passengers and pools increases, it may be hard to keep track of the passengers that exist in pools and those that do not, thus, the user might attempt to create a new pool with a passenger and be informed that that passenger already exists in a pool. The `findPool` feature thus allows the user to find the pool that the passenger already exists in to either confirm if they are in the correctly assigned pool, or `unpool` them if they are not. The `findPool` command works similarly to the `find` command, except that it currently only supports the use of the name prefix: "n/"
 
 The activity diagram below encapsulates the user workflow of attempting to pool a passenger, finding out that the passenger already exists in a pool, and using the `findPool` command to rectify the issue:
 
-![Activity Diagram for a user using Unpool](images/UnpoolActivityDiagram.png)
+![Activity Diagram for a user using Unpool](images/findPoolActivityDiagram.png)
```
Comment (chewterence) [inline docs/DeveloperGuide.md]: title of the activity diagram is still unpool, maybe change it to findpool?

**Review by chewterence — CHANGES_REQUESTED** [review]

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -260,6 +250,14 @@ From the diagram illustrated above:
 ### FindPool feature
 This feature allows users to find a pool that contains a passenger with a provided keyword in their name.
 
+Design considerations include the `findPool` command being able to be used in conjunction with the `pool` or `unpool`  command. Due to the limitations of the current iteration of GME, each passenger can only belong to one pool. As the number of passengers and pools increases, it may be hard to keep track of the passengers that exist in pools and those that do not, thus, the user might attempt to create a new pool with a passenger and be informed that that passenger already exists in a pool. The `findPool` feature thus allows the user to find the pool that the passenger already exists in to either confirm if they are in the correctly assigned pool, or `unpool` them if they are not. The `findPool` command works similarly to the `find` command, except that it currently only supports the use of the name prefix: "n/"
```
Comment (lyueyang) [inline docs/DeveloperGuide.md]: In
_the user might attempt to create a new pool with a passenger and be informed that that passenger already exists in a pool._ 

maybe it should be _that the_

**Review by lyueyang — CHANGES_REQUESTED** [review]

**Review by lyueyang — APPROVED** [review]
LGTM, issues highlighted by me have been corrected

**Pull request merged**